src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -130,17 +129,24 @@ protected override void Free()
             BoundNode node,
             ArrayBuilder<(BoundReturnStatement, TypeSymbolWithAnnotations)> returnTypesOpt,
             VariableState initialState,
-            Action<BoundExpression, TypeSymbolWithAnnotations> callbackOpt)
+            PooledDictionary<BoundExpression, TypeSymbolWithAnnotations> analyzedNullabilityMap = null)
             : base(compilation, method, node, new EmptyStructTypeCache(compilation, dev12CompilerCompatibility: false), trackUnassignments: true)
         {
-            _callbackOpt = callbackOpt;
             _binder = compilation.GetBinderFactory(node.SyntaxTree).GetBinder(node.Syntax);
             Debug.Assert(!_binder.Conversions.IncludeNullability);
             _conversions = (Conversions)_binder.Conversions.WithNullability(true);
             _useMethodSignatureReturnType = (object)methodSignatureOpt != null && useMethodSignatureReturnType;
             _useMethodSignatureParameterTypes = (object)methodSignatureOpt != null && useMethodSignatureParameterTypes;
             _methodSignatureOpt = methodSignatureOpt;
             _returnTypesOpt = returnTypesOpt;
+
+            if (analyzedNullabilityMap == null)
+            {
+                analyzedNullabilityMap = PooledDictionary<BoundExpression, TypeSymbolWithAnnotations>.GetInstance();
```

Comment:
>PooledDictionary<BoundExpression, TypeSymbolWithAnnotations>.GetInstance(); [](start = 41, length = 75)

The map should be allocated by the caller, or not allocated for scenarios where it's not needed.